packages/common/src/types/filterGrammar.ts

```diff
@@ -236,6 +252,9 @@ export const parseFilters = (
                         !!parsedFilter.is,
                     ),
                     values: parsedFilter.values || [1],
+                    settings: {
+                        unitOfTime: parsedFilter.date_interval,
+                    },
```

Comment:
```suggestion
                    ...(parsedFilter.date_interval
                        ? {
                              settings: {
                                  unitOfTime: parsedFilter.date_interval,
                              },
                          }
                        : {}),
```

that way you can remove the extra `undefined` additions in `filterGrammar.test.ts`

Reply:
done both, thanks